src/tokenizer/index.ts

```diff
@@ -212,12 +213,17 @@ export default class Tokenizer {
   }
 
 
-  private static getGuardianData (): Record<string, any> {
-    const item = localStorage.getItem('fp-guardian-results');
-    if (!item) { return [] }
-    const parsed = JSON.parse(item) as { expr: number; value: Record<string, any> }
-    if (!parsed || !parsed.value) { return [] }
-    return parsed.value
+  private getGuardianData (): Promise<{
+    events: Record<string, unknown>[],
+    session_id: string
+  }> {
+    const g = (window as {Guardian?: unknown}).Guardian as {getData?: () => Promise<{events: Record<string, unknown>[]; session_id: string;}>}
```

Comment:
Check the case if the window doesn't have Guardian. If it doesn't have it, then skip the operation at all.

Reply:
The following code snippet is checking exactly this and if window doesn't have guardian or contains guardian, but the guardian doesn't have a getData function implemented returns with `Promise.reject()` which will skip the setGuardian event emit and jumps into the finally block and submit the request.